traitcuration/static/js/mapping.js

```diff
@@ -27,14 +35,74 @@ function selectRow(row, tableId, traitId, termId) {
   selectedRow.classList.add("suggestion-table__row--selected");
 }
 
-// This function makes an ajax request to create a current mapping with the selected term 
+// This function makes an ajax request to create a current mapping with the selected term
 function mapButtonClicked() {
   if (selectedRowIndex < 1) return;
   axios
-    .post(`/traits/${currentTraitId}/mapping`, {'term': parseInt(selectedTermId)})
+    .post(`/traits/${currentTraitId}/mapping`, {
+      term: parseInt(selectedTermId),
+    })
     .then((response) => {
       // handle success
       console.log(response);
-      location.reload(); 
+      location.reload();
     });
 }
+
+
+/* This function accepts a term IRI pasted by a user and if it doesn't already exist, creates a new mapping suggestion
+and mapping for that term */
+async function existingTermButtonClicked() {
+  // Check if the IRI field is empty
+  termIRI = document.getElementById("existingTermIRI").value;
+  if (termIRI === "") return;
```

Comment:
Seems to me that this should also generate a warning popup, similarly to how it's done just below

Reply:
I will add that, yes.